src/main/java/persistence/context/PersistenceContext.java

```diff
@@ -0,0 +1,20 @@
+package persistence.context;
+
+import persistence.entity.EntitySnapshot;
+
+public interface PersistenceContext {
+
+    <T> T getEntity(Class<T> clazz, Object id);
+
+    void addEntity(Object entity);
+
+    void addEntity(Long id, Object entity);
+
+    void removeEntity(Object entity);
+
+    Object getDatabaseSnapshot(Long id, Object entity);
+
+    EntitySnapshot getCachedDatabaseSnapshot(Object entity);
+
+    EntitySnapshot getCachedDatabaseSnapshot(Long id, Object entity);
```

Comment:
getCachedDatabaseSnapshot 은 제가 강의 자료에 주석을 해놓았었는데요!!
실질적으로 특수하게 사용되는 메서드이긴 합니다 :)
보편적으로 사용되는 부분은 getDatabaseSnapshot 을 활용하고 있어요!! 메서드 네이밍만 변경해주면 좋을 것 같습니다

```suggestion
    Object getCachedDatabaseSnapshot(Long id, Object entity);

    EntitySnapshot getDatabaseSnapshot(Object entity);

    EntitySnapshot getDatabaseSnapshot(Long id, Object entity);
```

Reply:
@liquidjoo 아하 그렇군요. 수정하도록 하겠습니다👍